proposals/network_canister_management/ledger_20211117T2319Z.md

```diff
@@ -0,0 +1,124 @@
+// Subaccount is an arbitrary 32-byte byte array.
+// Ledger uses subaccounts to compute the source address, which enables one
+// principal to control multiple ledger accounts.
+type SubAccount = blob;
+
+// Sequence number of a block produced by the ledger.
+type BlockIndex = nat64;
+
+// An arbitrary number associated with a transaction.
+// The caller can set it in a `transfer` call as a correlation identifier.
+type Memo = nat64;
+
+// Arguments for the `transfer` call.
+type TransferArgs = record {
+    // Transaction memo.
+    // See comments for the `Memo` type.
+    memo: Memo;
+    // The amount that the caller wants to transfer to the destination address.
+    amount: ICP;
+    // The amount that the caller pays for the transaction.
+    // Must be 10000 e8s.
+    fee: ICP;
+    // The subaccount from which the caller wants to transfer funds.
+    // If null, the ledger uses the default (all zeros) subaccount to compute the source address.
+    // See comments for the `SubAccount` type.
+    from_subaccount: opt SubAccount;
+    // The destination account.
+    // If the transfer is successful, the balance of this address increases by `amount`.
+    to: AccountIdentifier;
+    // The point in time when the caller created this request.
+    // If null, the ledger uses current IC time as the timestamp.
+    created_at_time: opt Timestamp;
+};
+
+type TransferError = variant {
+    // The fee that the caller specified in the transfer request was not the one that ledger expects.
+    // The caller can change the transfer fee to the `expected_fee` and retry the request.
+    BadFee : record { expected_fee : ICP; };
+    // The account specified by the caller doesn't have enough funds.
+    InsufficientFunds : record { balance: ICP; };
+    // The request is too old.
+    // The ledger only accepts requests created within 24 hours window.
+    // This is a non-recoverable error.
+    TxTooOld : record { allowed_window_nanos: nat64 };
+    // The caller specified `created_at_time` that is too far in future.
+    // The caller can retry the request later.
+    TxCreatedInFuture : null;
+    // The ledger has already executed the request.
+    // `duplicate_of` field is equal to the index of the block containing the original transaction.
+    TxDuplicate : record { duplicate_of: BlockIndex; }
+};
+
+type TransferResult = variant {
+    Ok : BlockIndex;
+    Err : TransferError;
+};
+
+// Arguments for the `account_balance` call.
+type AccountBalanceArgs = record {
+    account: AccountIdentifier;
+};
+
+service : {
+  // Transfers tokens from a subaccount of the caller to the destination address.
+  // The source address is computed from the principal of the caller and the specified subaccount.
+  // When successful, returns the index of the block containing the transaction.
+  transfer : (TransferArgs) -> (TransferResult);
+
+  // Returns the amount of ICP on the specified account.
+  account_balance : (AccountBalanceArgs) -> (ICP) query;
+}
```

Comment:
```suggestion
}
```
I'm not familiar with the semicolon policy of candid, do you need one to terminate here?

Reply:
In Candid, semicolon is not a terminator, but a separator. Though it's allowed to end a sequence with a semicolon.